octodns_constellix/__init__.py

```diff
@@ -20,78 +20,114 @@
         self.api_key = api_key
         self.secret_key = secret_key
         self.ratelimit_delay = ratelimit_delay
         self._sess = Session()
         self._sess.headers.update(
             {
-                'x-cnsdns-apiKey': self.api_key,
-                'User-Agent': f'octodns/{octodns_version} octodns-constellix/{__VERSION__}',
+                'User-Agent': f'octodns/{octodns_version} octodns-constellix/{__VERSION__}'
             }
         )
-        self._domains = None
-        self._pools = {'A': None, 'AAAA': None, 'CNAME': None}
-        self._geofilters = None
 
     def _current_time(self):
         return str(int(time.time() * 1000))
 
-    def _hmac_hash(self, now):
-        return hmac.new(
-            self.secret_key.encode('utf-8'),
-            now.encode('utf-8'),
-            digestmod=hashlib.sha1,
-        ).digest()
+    def _hmac_text(self, now):
+        return b64encode(
+            hmac.new(
+                self.secret_key.encode('utf-8'),
+                now.encode('utf-8'),
+                digestmod=hashlib.sha1,
+            ).digest()
+        )
+
+    def _url(self, path):
+        return f'{self.base_url}{path}'
+
+    def _get_json(self, response):
+        try:
+            return response.json()
+        except:
+            raise ConstellixAPIBadRequest({'errors': [response.text]})
 
     def _request(self, method, path, params=None, data=None):
-        self.log.debug("Call _request %s %s", method, path)
+        url = self._url(path)
+        self.log.debug('Call _request %s %s', method, url)
         now = self._current_time()
-        hmac_hash = self._hmac_hash(now)
+        hmac_text = self._hmac_text(now)
+        auth_token = f'{self.api_key}:{hmac_text}:{now}'
 
         headers = {
-            'x-cnsdns-hmac': b64encode(hmac_hash),
-            'x-cnsdns-requestDate': now,
+            'x-cns-security-token': auth_token,  # v1, v2, sonar
+            'authorization': f'Bearer {auth_token}',  # v4
         }
 
-        url = f'{self.BASE}{path}'
         resp = self._sess.request(
             method, url, headers=headers, params=params, json=data
         )
-        if resp.status_code == 400:
-            raise ConstellixClientBadRequest(resp)
-        if resp.status_code == 401:
-            raise ConstellixClientUnauthorized()
-        if resp.status_code == 404:
-            raise ConstellixClientNotFound()
+
+        status_code = resp.status_code
+        headers = resp.headers
+
+        if status_code == 400:
+            raise ConstellixAPIBadRequest(self._get_json(resp))
+        if status_code == 401:
+            raise ConstellixAPIUnauthorized()
+        if status_code == 404:
+            raise ConstellixAPINotFound()
         resp.raise_for_status()
+
+        if self.ratelimit_delay >= 1.0:
+            self.log.info('Waiting for Constellix Limit Delay')
+        elif self.ratelimit_delay > 0.0:
+            self.log.debug('Waiting for Constellix Rate Limit Delay')
```

Comment:
What's the difference in > 1 and > 0 here. Messages seem fairly similar and i see there's a log level difference too.

Reply:
This is old code from the `SonarClient`.

It has been there since the beginning of this repo (1f4f4952ff164b2e3dbe3aa2ba89994f9fc0a588), so I just kept it.

We might as well get rid of it, as it does not make much sense. Probably the idea was to warn the user if the slowdown was significant.

Reply:
Addressed in c27d688a736d7406cb615fed72edd32e80d637c1